python/cudf/cudf/core/dataframe.py

```diff
@@ -2539,6 +2539,19 @@ def groupby(
             raise TypeError(
                 "groupby() requires either by or level to be" "specified."
             )
+        df = self
+        if isinstance(by, Series):
```

Comment:
We should be able to group by arbitrary sequences (not just `Series` objects), so I'm not sure if this is the best place to do this. Inside the groupby code somewhere we convert `by` to a `Series` and that's where this should happen.

Reply:
I take this back. It looks like Pandas only allows grouping by `Series` if the length of the grouper is different from the dataframe

Reply:
And if it is a sequence, it normally expects it to be columns names

Reply:
Hmm - this still won't do. In general, you can group by multiple `Series` objects, or a combination of columns and `Series` objects:

```
In [64]: a = pd.DataFrame({'a': [1, 1, 1, 2, 2], 'b': [1, 2, 3, 4, 5]})                                                                                                                                         

In [65]: a.groupby(['a', pd.Series([1, 2, 3])])                                                                                                                                                                 
Out[65]: <pandas.core.groupby.generic.DataFrameGroupBy object at 0x7f9097a0a400>
```

Reply:
I didn't realize this scenario, then I have handle it as you had mentioned @ https://github.com/rapidsai/cudf/blob/b96073c7b481087f009849eee527cfe78ac8c081/python/cudf/cudf/core/groupby/groupby.py#L201